bpf/lib/nodeport.h

```diff
@@ -1019,24 +1019,35 @@ nodeport_rev_dnat_fwd_ipv6(struct __ctx_buff *ctx, struct trace_ctx *trace)
 		return ret;
 	}
 
-	if (!ct_has_nodeport_egress_entry6(get_ct_map6(&tuple), &tuple))
+	if (!ct_has_nodeport_or_dsr_egress_entry6(get_ct_map6(&tuple), &tuple))
 		return CTX_ACT_OK;
 
 	ret = ct_lookup6(get_ct_map6(&tuple), &tuple, ctx, l4_off, CT_INGRESS,
 			 &ct_state, &trace->monitor);
 
-	if (ret == CT_REPLY && ct_state.node_port && ct_state.rev_nat_index) {
-		struct csum_offset csum_off = {};
-
+	if (ret == CT_REPLY) {
 		trace->reason = TRACE_REASON_CT_REPLY;
-		csum_l4_offset_and_flags(tuple.nexthdr, &csum_off);
 
-		ret = lb6_rev_nat(ctx, l4_off, &csum_off, ct_state.rev_nat_index,
-				  &tuple, REV_NAT_F_TUPLE_SADDR);
-		if (IS_ERR(ret))
-			return ret;
+		if (ct_state.node_port && ct_state.rev_nat_index) {
+			struct csum_offset csum_off = {};
 
-		ctx_snat_done_set(ctx);
+			csum_l4_offset_and_flags(tuple.nexthdr, &csum_off);
+
+			ret = lb6_rev_nat(ctx, l4_off, &csum_off, ct_state.rev_nat_index,
+					  &tuple, REV_NAT_F_TUPLE_SADDR);
+			if (IS_ERR(ret))
+				return ret;
+
+			ctx_snat_done_set(ctx);
```

Comment:
You can perhaps put this one line 1050, and remove the one line 1028?

Reply:
Not sure I understand - you mean moving this into the general `if (ret == CT_REPLY)` case? That would mean potentially flagging a packet as SNAT-done even if we didn't do any SNAT processing, no?